src/duplicacy_snapshot.go

```diff
@@ -76,14 +76,24 @@ func CreateSnapshotFromDirectory(id string, top string) (snapshot *Snapshot, ski
                 continue
             }
 
-            if pattern[0] != '+' && pattern[0] != '-' {
+            if pattern[0] == '#' {
+                continue
+            }
+
+            if IsUnspecifiedFilter(pattern) {
                 pattern = "+" + pattern
             }
 
-            if pattern == "+" || pattern == "-" {
+            if IsEmptyFilter(pattern) {
                 continue
             }
 
+            valid, err := IsValidRegex(pattern)
```

Comment:
Doesn't this line demand that all patterns be regexes? That's not what we want.

Reply:
oops, yes, has been fixed in latest PR.